piqueserver/config.py

```diff
@@ -0,0 +1,145 @@
+# Copyright (c) 2017 Piqueserver development team
+
+# This file is part of piqueserver.
+
+# piqueserver is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# piqueserver is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with piqueserver.  If not, see <http://www.gnu.org/licenses/>.
+
+import json
+import toml
+
+
+DEFAULT_STYLE = 'TOML'
+TOML_STYLE = 'TOML'
+JSON_STYLE = 'JSON'
+
+
+class ConfigException(Exception):
+    pass
+
+
+class ConfigStore():
+    '''
+    configuration store that manages global configuration
+    '''
+
+    def __init__(self):
+        self._raw_config = {}
+        self.options = {}
+
+    def validate_all(self):
+        for option in self.options.values():
+            option.validate(option.get())
+
+    def clear_config(self):
+        self._raw_config = {}
+
+    def load_config(self, config_file, style=DEFAULT_STYLE):
```

Comment:
the config part of the name is redundant, maybe replace it with some more useful information like `load_from_file()`